wscapableproxy.go

```diff
@@ -102,13 +101,28 @@ func (p *WebsocketCapableReverseProxy) ServeWebsocket(inConn *websocket.Conn, r
 		if resp != nil {
 			log.Printf("outbound websocket dial error, status: %v, err: %v",
 				resp.StatusCode, err)
+			w.WriteHeader(resp.StatusCode)
+			_, err := io.Copy(w, resp.Body)
+			if err != nil {
+				log.Printf("error copying outbound body to response. err: %v", err)
+			}
 		} else {
 			log.Printf("outbound websocket dial error, err: %v", err)
+			w.WriteHeader(502)
+			fmt.Fprintln(w, "Bad Gateway")
```

Comment:
The usual convention is to say `http.Error(w, "502 Bad Gateway", http.StatusBadGateway)`.